packages/typespec-ts/test/util/emitUtil.ts

```diff
@@ -377,13 +376,15 @@ export async function emitModularSerializeUtilsFromTypeSpec(
         casing: "camel"
       }
     );
-    if (
-      modularCodeModel &&
-      modularCodeModel.clients &&
-      modularCodeModel.clients.length > 0 &&
-      modularCodeModel.clients[0]
-    ) {
-      return buildSerializeUtils(modularCodeModel);
+    if (modularCodeModel.clients[0]) {
+      buildSerializers(
+        dpgContext,
+        modularCodeModel,
+        modularCodeModel.clients[0]
+      );
+      return modularCodeModel.project
+        .getSourceFiles()
+        .filter((sf) => sf.getFilePath().endsWith("serializeUtil.ts"));
```

Comment:
it feels a little strange to get the SourceFile this way, can we split the buildSerializers into two functions one can build and return the serializer, the other can return the source file?